### DISCUSSION TITLE
DFR0971

### DISCUSSION THREAD

**Discussion by ThomasKlaus-ai** [arendst/Tasmota#20810] [Q&A] [answered]
Ich habe zu meinem ESP32 einen DFR0971 über I2C angeschlossen. Gibt es eine Möglichkeit den DAC über Tasmota in mein IoBroker einzubinden ? Ich kann zwar I2C in Tasmota einstellen und kann in der Konsole auch die Adresse über den I2C Scan finden, aber ich habe keinen Datenpunkt in IoBroker um mit Blockly arbeiten zu können.

**Comment by HofeBY** [comment]
Did you find DFR0971 in Sourcetree ./tasmota/tasmota/my_user_config.h ?

If not included, offer your driver .

**Reply by ThomasKlaus-ai** [reply]
Sorry, but I don‘t know how to finde the sourcetree in my system.But in the output pin, I can only set I2C SDA and SCL in Tasmota. Nevertheless, I don't see a value for the voltage output.

**Reply by Jason2866** [reply, marked as answer]
DFR0971 is not a supported device from Tasmota. You can't use with Tasmota.
For using "someone" has to write a driver for Tasmota.

**Reply by HofeBY** [reply]
No wonder not seeing any value like Voltage.
Driver is not existing until now.

Sourcetree can be downloaded on site
 https://github.com/arendst/Tasmota now click on "green button (Click)" then "Download Zip"
after downlod unzip to any folder an search throug ./tasmota/tasmota/my_user_config.h
In this file you must find your seached driver. If not, write one.

greetings from Oberfranken

**Reply by ThomasKlaus-ai** [reply]
Thank you for the info. Since I'm unfortunately not so fit to write a driver, I'll probably have to wait for it. Maybe at some point there will be a possibility. I also have a code with the IDE, but I don't know how to create a data point in IoBroker with MQTT.

**Reply by sfromis** [reply]
With the Berry language, Tasmota does have support for the built-in DAC channels of ESP32, without needing an extra chip. Berry is also an easier avenue to coding a driver for an I2C device like you have there.

**Comment by Jason2866** [comment]
This discussion is English only. Please translate your question

**Reply by barbudor** [reply]
Actually DFR0971 is the DFRobot reference of the module based on GP8403, a 12-bit DAC from chinese company GuestGood
Datasheet only available in chinese

That doesn't change anything to the point that GP8403 is not (yet) supported by Tasmota

As you are using en ESP32, you can use the built-in 8-bit DAC of the ESP32 which can be controlled in Berry language
https://tasmota.github.io/docs/Berry/#dac-gpios

If you absolutely need 12-bit, then you will have to write the driver for that GP8403 (or preferably something less exotic such as MCP4725)
